packages/openapi-generator/src/file-serializer/interface-file.ts

```diff
@@ -0,0 +1,37 @@
+import { codeBlock } from '@sap-cloud-sdk/util';
+import { OpenApiNamedSchema, OpenApiSchema } from '../openapi-types';
+import {
+  collectRefs,
+  hasNotSchema,
+  parseTypeNameFromRef,
+  parseFileNameFromRef
+} from '../model';
+import { serializeSchema } from './schema';
+import { Import, serializeImports } from './imports';
+
+/**
+ * Serialize a schema representation to a string representing the according interface file contents.
+ * @param operationInfo A named schema.
+ * @returns The serialized interface file contents.
+ */
+export function interfaceFile({ name, schema }: OpenApiNamedSchema): string {
+  const imports = serializeImports(getImports(schema));
+
+  return codeBlock`
+    ${imports}
+    export type ${name} = ${serializeSchema(schema)};
+  `;
+}
+
+function getImports(schema: OpenApiSchema): Import[] {
```

Comment:
The `getImports` method feels a bit hacky. What about that the `serializeSchema` does not only returns a TypeScript sting but also a list all depedencies which are required but this string to work.

Reply:
Discussed, rather minor and brings other complexity.